cm-server/src/main/java/ro/msg/cm/controller/UpdateStudyYearController.java

```diff
@@ -0,0 +1,178 @@
+        this.startYearProperties = startYearProperties;
+    }
+
+    @RequestMapping(value = "/candidates", method = RequestMethod.GET, produces = "application/hal+json")
+    public Resources updateUniversityYear() throws ParseException {
+
+        Iterable<Candidate> candidates = findAll();
+        List<Candidate> candidateList = new ArrayList<>();
+        candidates.forEach(candidateList::add);
+        List<Resource<Candidate>> resources = new LinkedList<>();
+        for (Candidate candidate : candidates) {
+            candidate.setCurrentStudyYear(determineYearBasedOnDuration(candidate));
+            resources = candidateToResource(candidateList);
+        }
+        return new Resources<>(resources);
+
+    }
+
+    private Resource<Candidate> getResource(Candidate candidate) throws ParseException {
+
+        Resource<Candidate> resource = new Resource<>(candidate);
+        resource.add(linkTo(methodOn(UpdateStudyYearController.class).getResourceCandidateById(candidate.getId())).withSelfRel());
+        resource.add(linkTo(methodOn(UpdateStudyYearController.class).getResourceCandidateById(candidate.getId())).withRel("candidate"));
+        resource.add(linkTo(methodOn(CandidateSkillsController.class).getCandidateSkills(candidate.getId())).withRel("candidateSkillsList"));
+        resource.add(linkTo(methodOn(EducationController.class).getEducation(candidate.getId())).withRel("education"));
+        resource.add(linkTo(methodOn(CandidateNotesController.class).getCandidateNotes(candidate.getId())).withRel("candidateNotesList"));
+
+        return resource;
+    }
+
+    private List<Resource<Candidate>> candidateToResource(List<Candidate> candidates) throws ParseException {
+        List<Resource<Candidate>> resources = new ArrayList<>(candidates.size());
+        for (Candidate candidate : candidates) {
+            Resource<Candidate> resource = getResource(candidate);
+            resources.add(resource);
+        }
+        return resources;
+    }
+
+    @RequestMapping(value = "candidates/{id}", method = RequestMethod.GET, produces = "application/hal+json")
+    public Resource getResourceCandidateById(@PathVariable(value = "id") Long id) throws ParseException {
+
+        Candidate candidate = candidateRepository.findOne(id);
+        Resource<Candidate> resource = new Resource<>(candidate);
+        resource.add(linkTo(methodOn(UpdateStudyYearController.class).getResourceCandidateById(id)).withRel("candidate"));
+        return resource;
+    }
+
+    private Iterable<Candidate> findAll() {
+            return candidateRepository.findAll();
+    }
+
+    /**
+     * Method that calculate the current study year based on the duration of study
+     *
+     * @param candidate Candidate
+     * @return int - the current study year
+     * @throws ParseException indicates if errors occurred
+     */
+    private int determineYearBasedOnDuration(Candidate candidate) throws ParseException {
```

Comment:
This method, along with all the related ones, I would move into a "stateful bean" (an own class, annotated with a `@Component ` + `@Scope` annotation ). The bean would have as a instance attributes the current date / calendar, maybe the SimpleDateFormatter (such that we avoid instantiating them in a loop) and the properties. It is important that this bean is instantiated newly on each call. See https://stackoverflow.com/a/33743026/7612556